x/mint/keeper/keeper_test.go

```diff
@@ -61,49 +65,67 @@ func (suite *KeeperTestSuite) setupDeveloperVestingModuleAccountTest(blockHeight
 	}
 }
 
-func TestKeeperTestSuite(t *testing.T) {
-	suite.Run(t, new(KeeperTestSuite))
-}
-
-func (suite *KeeperTestSuite) TestMintCoinsToFeeCollectorAndGetProportions() {
+// TestGetProportions tests that mint allocations are computed as expected.
+func (suite *KeeperTestSuite) TestGetProportions() {
 	tests := []struct {
-		name                 string
-		ratio                sdk.Dec
-		hasPreExistingSupply bool
-		expectedCoin         sdk.Coin
-		fee                  sdk.Coin
+		name          string
+		ratio         sdk.Dec
+		expectedCoin  sdk.Coin
+		expectedError error
+		mintedCoin    sdk.Coin
 	}{
 		{
-			name:                 "coin is minted to the fee collector",
-			fee:                  sdk.NewCoin("stake", sdk.NewInt(0)),
-			ratio:                sdk.NewDecWithPrec(2, 1),
-			hasPreExistingSupply: false,
-			expectedCoin:         sdk.NewCoin("stake", sdk.NewInt(0)),
-		}, {
-			name:                 "mint the 100K stake coin to the fee collector",
-			fee:                  sdk.NewCoin("stake", sdk.NewInt(100000)),
-			ratio:                sdk.NewDecWithPrec(2, 1),
-			hasPreExistingSupply: true,
-			expectedCoin:         sdk.NewCoin("stake", sdk.NewInt(100000).Quo(sdk.NewInt(5))),
+			name:         "0 * 0.5 = 0",
+			mintedCoin:   sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(0)),
+			ratio:        sdk.NewDecWithPrec(2, 1),
+			expectedCoin: sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(0)),
+		},
+		{
+			name:         "100000 * 0.2 = 20000",
+			mintedCoin:   sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100000)),
+			ratio:        sdk.NewDecWithPrec(2, 1),
+			expectedCoin: sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100000).Quo(sdk.NewInt(5))),
+		},
+		{
+			name:         "123456 * 2/3 = 82304",
+			mintedCoin:   sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(123456)),
+			ratio:        sdk.NewDecWithPrec(2, 1).Quo(sdk.NewDecWithPrec(3, 1)),
+			expectedCoin: sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(82304)),
+		},
+		{
+			name:       "54617981 * 131/273 approx = 2.62",
+			mintedCoin: sdk.NewCoin("uosmo", sdk.NewInt(54617981)),
+			ratio:      sdk.NewDecWithPrec(131, 3).Quo(sdk.NewDecWithPrec(273, 3)),
```

Comment:
To be pedantic, this is then a `.131 / .273` test (which may differ in rounding precision)

Thoughts on extracting this to a var, to simplify the expected coin expression?

Reply:
I missed the dots in front but the intention was to have these values. I'm not sure I understand the concern about precision. I specified "approx =" in the test name indicating the potential rounding difference. Is there anything else I'm missing? 

As for the var, extracted it to `complexRatioDec`, thanks for the suggestions!

Reply:
I'm going to merge this. If there are follow-up items that need to be addressed, happy to do so in another PR